### PR TITLE
fix(security): sanitize host_id and service_id - 2.8.x

### PR DESCRIPTION
**Pull request by lpinsivy**
# Description

Sanitize host_id and service_id

Based on #7862 but for 2.8.x compatibility

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [x] 2.8.x
- [ ] 18.10.x
- [ ] 19.04.x
- [ ] 19.10.x (master)

#### Community contributors & Centreon team

- [x] I followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

#### Centreon team only

- [x] I have made sure that the **unit tests** related to the story are successful.
- [x] I have made sure that **unit tests cover 80%** of the code written for the story.
- [ ] I have made sure that **acceptance tests** related to the story are successful (**local and CI**)

### CODE REVIEW

File: www/include/monitoring/status/Services/xml/makeXMLForOneHost.php [hunk 1]
```diff
@@ -82,6 +83,27 @@
 $disable        = $obj->checkArgument("disable", $_GET, "disable");
 $dateFormat     = $obj->checkArgument("date_time_format_status", $_GET, "Y/m/d H:i:s");
 
+$host_id = filter_var(
+    $host_id ? $host_id : null,
```
Comment (sc979) [inline www/include/monitoring/status/Services/xml/makeXMLForOneHost.php]:
```suggestion
    isset($host_id) ? $host_id : null,
```

File: www/include/monitoring/status/Services/xml/makeXMLForOneService.php [hunk 2]
```diff
@@ -83,8 +84,31 @@
 $dateFormat         = $obj->checkArgument("date_time_format_status", $_GET, "Y/m/d H:i:s");
 
 $tab = preg_split('/\_/', $svc_id);
-$host_id = $tab[0];
-$service_id = $tab[1];
+$host_id = filter_var(
+    $tab[0] ? $tab[0] : null,
```
Comment (sc979) [inline www/include/monitoring/status/Services/xml/makeXMLForOneService.php]:
```suggestion
    isset($tab[0]) ? $tab[0] : null,
```

File: www/include/monitoring/status/Services/xml/makeXMLForOneService.php [hunk 3]
```diff
@@ -83,8 +84,31 @@
...
+    FILTER_VALIDATE_INT
+);
+
+$service_id = filter_var(
+    $tab[1] ? $tab[1] : null,
```
Comment (sc979) [inline www/include/monitoring/status/Services/xml/makeXMLForOneService.php]:
```suggestion
    isset($tab[1]) ? $tab[1] : null,
```

**Comment by kduret** [comment]
If you print a message, instead of returning a XML, how the web page reacts ?

File: www/include/monitoring/status/Services/xml/makeXMLForOneHost.php [hunk 4]
```diff
@@ -82,6 +83,27 @@
 $disable        = $obj->checkArgument("disable", $_GET, "disable");
 $dateFormat     = $obj->checkArgument("date_time_format_status", $_GET, "Y/m/d H:i:s");
 
+$host_id = filter_var(
+    isset($host_id) ? $host_id : null,
+    FILTER_VALIDATE_INT
+);
+
+if (!$host_id) {
```
Comment (callapa) [inline www/include/monitoring/status/Services/xml/makeXMLForOneHost.php]:
```suggestion
if ($host_id === false) {
```

File: www/include/monitoring/status/Services/xml/makeXMLForOneService.php [hunk 5]
```diff
@@ -83,8 +84,31 @@
...
+    isset($tab[0]) ? $tab[0] : null,
+    FILTER_VALIDATE_INT
+);
+
+$service_id = filter_var(
+    isset($tab[1]) ? $tab[1] : null,
+    FILTER_VALIDATE_INT
+);
+
+if (!$host_id || !$service_id) {
```
Comment (callapa) [inline www/include/monitoring/status/Services/xml/makeXMLForOneService.php]:
```suggestion
if ($host_id === false || $service_id === false) {
```

**Pull request merged**